src/components/smart_objects/src/array_schema_item.cc

```diff
@@ -90,6 +92,31 @@ errors::eType CArraySchemaItem::validate(
   return errors::OK;
 }
 
+bool CArraySchemaItem::filterInvalidEnums(
```

Comment:
I feel this function as rather complicated one. 
I would try probably to sacrifice performance but make it more easy to read and obvious.

Reply:
Perhaps I might be able to add some extra comments for clarity. I don't think there's really a way to make it simpler though

Reply:
Added comments in https://github.com/smartdevicelink/sdl_core/pull/3385/commits/147b7bca79fe459cdceeeee6d7b8a74133890571